examples/structured_data/deep_neural_decision_forests.py

```diff
@@ -0,0 +1,447 @@
+"""
+Title: Classification with Neural Decision Forests
+Author: [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/)
+Date created: 2021/01/15
+Last modified: 2021/01/15
+Description: How to train differentiable decision trees for end-to-end learning in deep neural networks.
+"""
+
+"""
+## Introduction
+
+This example provides an implementation of the
+[Deep Neural Decision Forest](https://ieeexplore.ieee.org/document/7410529)
+model, introduced by P. Kontschieder et al. for structured data classification.
+This model introduces a stochastic and differentiable decision tree model,
+trained end-to-end, unifying decision trees with deep representation learning.
+
+## The dataset
+
+This example uses the
+[United States Census Income Dataset](https://archive.ics.uci.edu/ml/datasets/census+income)
+provided by the
+[UC Irvine Machine Learning Repository](https://archive.ics.uci.edu/ml/index.php).
+The task is binary classification
+to predict whether a person is likely to be making over USD 50k a year.
+
+The dataset includes 48,842 instances with 14 input features: 5 numerical features
```

Comment:
Suggestion: just adding just a little bit more information about the dataset.

```suggestion
The dataset includes 48,842 instances with 14 input features (such as age, work class, education, occupation, and so on): 5 numerical features
```